build_rocm

```diff
@@ -22,6 +22,6 @@ rm -f $TF_PKG_LOC/tensorflow*.whl
 
 yes "" | TF_NEED_ROCM=1 PYTHON_BIN_PATH=/usr/bin/python ./configure
 pip uninstall -y tensorflow || true
-bazel build --config=opt --config=rocm --action_env=HIP_PLATFORM=hcc //tensorflow/tools/pip_package:build_pip_package --verbose_failures &&
+bazel build --config=opt --config=mkl --copt="-O3" --copt=-mavx2 --copt=-mfma --copt=-mfpmath=both --config=rocm --action_env=HIP_PLATFORM=hcc //tensorflow/tools/pip_package:build_pip_package --verbose_failures &&
```

Comment:
@whchung 

Would this  (https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/blob/develop-upstream/third_party/gpus/crosstool/BUILD.rocm.tpl#L76) be a better place to put all the `--copt` options?

Also would it make more sense to split this into two PRs?
1. add the `--config=mkl` option to ROCm build
2. add the various `--copt` options to ROCm build

Reply:
@deven-amd I don't know if hiding these details in `BUILD.rocm.tpl` be a good idea. That file should contain the very default options. And in this PR it seems we are proposing some release-specific configurations.